frontend/lib/src/components/shared/WindowDimensions/index.tsx

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createContext } from "react"
+
+export type WindowDimensions = {
+  fullWidth: number
+  fullHeight: number
+}
```

Comment:
nit: I believe we usually just re-export in `index.tsx` files and do not define any new types or similar in there. So perhaps we should adhere to the existing style and move this into the `Provider.tsx` or the `useWindowDimensions.tsx`